src/solidity/typeDecoder.js

```diff
@@ -0,0 +1,288 @@
+'use strict'
+
+/**
+  * determine if the given @arg type need to start from a new slot
+  *
+  * @param {Object} type - current type (object returned by the getType function)
+  * @return true if the type needs a new slot
+  */
+function typeNeedNewSlot (type) {
+  return (type.storageBytes === 'dynamic' || type.arraySize || type.members || type.fullType === 'string' || type.fullType === 'bytes')
```

Comment:
Please don't use `type.arraySize` like that. Furthermore, this should actually be a polymorphic function of the type.